### PR TITLE
`nixpkgs` upgrade to `23.11`

### LINKED ISSUE #18311: Upgrade nixpkgs to release `23.11`
A `nixpkgs` upgrade is available `23.11` -> https://github.com/NixOS/nixpkgs/releases/tag/23.11

This solves our requirement to upgrade `golang` beyond `1.20` 

changes form our existing 23.05 -> https://github.com/NixOS/nixpkgs/compare/23.05...23.11

### PR DESCRIPTION
**Pull request by siddarthkay**
## Summary

This PR upgrades `nixpkgs` to `23.11` 

We also apply fixes from https://github.com/NixOS/nixpkgs/pull/269479 in our fork of `nixpkgs` and use that commit.

## Depends on 

- [ ] https://github.com/status-im/status-mobile/pull/18564 for iOS related fixes (commit is included in this PR as well)
- [x] https://github.com/status-im/status-mobile/pull/18563 for android related fixes (commit is included in this PR as well)

## Notable changes from previous version 

- [ ] `Gradle` has been bumped from `8.0.1` to `8.4`
- [ ] `clojure-lsp` was upgraded
- [ ] add more stuff

fixes #18311 

## Platforms

- Android
- iOS

status: ready

### CODE REVIEW

**Comment by jakubgs** [comment]
You can't find `nodejs-20.x` because it doesn't exist:
```
~/work/nixpkgs remotes/origin/clevis~326
 > g rev-parse HEAD                                         
ddf0003c57fb5cbb4a9754f2f6d5ebe9cdae5151

~/work/nixpkgs remotes/origin/clevis~326
 > grep 'nodejs_[0-9]\+ =' pkgs/top-level/all-packages.nix  
  nodejs_18 = callPackage ../development/web/nodejs/v18.nix { };
  nodejs_20 = callPackage ../development/web/nodejs/v20.nix { };
  nodejs_21 = callPackage ../development/web/nodejs/v21.nix { };

~/work/nixpkgs remotes/origin/clevis~326
 > grep 'nodejs-[0-9]\+' pkgs/top-level/aliases.nix       
  nodejs-14_x = nodejs_14; # Added 2022-11-06
  nodejs-16_x = nodejs_16; # Added 2022-11-06
  nodejs-16_x-openssl_1_1 = throw "nodejs-16_x-openssl_1_1 has been removed."; # Added 2023-02-04
  nodejs-18_x = nodejs_18; # Added 2022-11-0
```
Use `nodejs_20`.

Also, the commit your using this, where is it from?

**Comment by jakubgs** [comment]
Also, it's important to use stable branches like `nixos-23.11`, and not commits from `master`:

image

- https://github.com/NixOS/nixpkgs/commit/ddf0003c57fb5cbb4a9754f2f6d5ebe9cdae5151

**Comment by jakubgs** [comment]
Here's an example of a `nixpkgs` upgrade for reference:

* https://github.com/status-im/status-mobile/pull/14944

It is good to list notable upgrade in the commit and PR description to make it clear what is being changes in tooling and libraries.

**Comment by jakubgs** [comment]
It appears @richard-ramos needs Go 1.21 and for that reason is interested in upgrading `nixpkgs` used by `status-go`. But to maintain predictable behavior we should be using the same `nixpkgs` as we do in mobile and desktop.

Please coordinate with him to settle on the same version, preferably from stable `nixos-23.11` branch.

**Comment by siddarthkay** [comment]
converting to draft since this will require lots of ongoing work

File: ios/Podfile
Comment (jakubgs) [inline ios/Podfile:72]: Wrong indent. Also no comment explaining why this is done. Just a RELATIVE link that targets `main` branch, which will become useless soon.

File: nix/deps/gradle/generate.sh
Comment (jakubgs) [inline nix/deps/gradle/generate.sh:57]: This is so stupid.

File: nix/pkgs/xcbeautify/default.nix
Comment (jakubgs) [inline nix/pkgs/xcbeautify/default.nix:13]: Why did the repo owner change?

File: nix/status-go/mobile/build.nix
Comment (jakubgs) [inline nix/status-go/mobile/build.nix:50]: Why was this flag added?

**Review by jakubgs — CHANGES_REQUESTED** [review]
Why is the upgrade to RN 0.73.2 part of this PR?

Also, the PR description does not contain a list of major upgrades of tools and libraries included in this `nixpkgs` ugprade.

I would also keep the linter changes as a separate commit when squashing this.

File: nix/pkgs/xcbeautify/default.nix
Comment (yakimant) [reply nix/pkgs/xcbeautify/default.nix:13]: As I see it was moved to a new owner. Previous link still works - it redirects.

File: nix/pkgs.nix
Comment (yakimant) [inline nix/pkgs.nix:10]: Maybe taking from 23.11 branch head is better, it can contain usefull fixes.

File: nix/status-go/mobile/build.nix
Comment (siddarthkay) [reply nix/status-go/mobile/build.nix:50]: to fix : https://github.com/golang/go/issues/58416 which was discovered in this PR https://github.com/status-im/status-mobile/pull/18564 and iOS build here were also failing with the same issue

**Comment by siddarthkay** [comment]
> Why is the upgrade to RN 0.73.2 part of this PR?

because `nixpkgs` upgrades `gradle` to `8.4` and this `gradle` version was too high for `react-native` version in develop i.e `0.72.5`
However when the latest `react-native` version `0.73.2` works well with `gradle` 8.4

**Comment by siddarthkay** [comment]
I also hope to merge this PR after the RN upgrade has been merged, so the RN upgrade commit in this PR would go away with rebase.

File: nix/status-go/mobile/build.nix
Comment (jakubgs) [reply nix/status-go/mobile/build.nix:50]: I see, i wasn't aware this was from another PR. This is what happens when you create a lot of PRs that depend on one another and are rebased on top of one another.

Comment (siddarthkay) [reply nix/status-go/mobile/build.nix:50]: yeah sorry for this confusion. 
I usually never do this, I was trying to get CI pipelines to be green for `nixpkgs` upgrade 😿 
I did try to mention it in PR description though 😅

**Comment by siddarthkay** [comment]
update: will resume working on this after the release is done.

**Comment by siddarthkay** [comment]
This would become a tough rebase.
closing this PR in favour of new PR -> https://github.com/status-im/status-mobile/pull/19369

**Pull request closed without merging**
